--- clamav-0.102.2/clamd/server-th.c	2020-02-12 15:05:02.000000000 -0500
+++ clamav-0.102.2/clamd/server-th.c.orig	2020-02-12 15:01:45.060887210 -0500
@@ -1171,7 +1171,7 @@
     mainpid = getpid();
     if ((opt = optget(opts, "PidFile"))->enabled) {
         FILE *fd;
-        old_umask = umask(0002);
+        old_umask = umask(0022);
         if ((fd = fopen(opt->strarg, "w")) == NULL) {
             logg("!Can't save PID in file %s\n", opt->strarg);
         } else {
