--- a/mk/kernel.mk	2019-06-10 10:22:04.000000000 -0400
+++ b/mk/kernel.mk+	2019-11-19 17:16:26.190663609 -0500
@@ -380,24 +380,24 @@
 # module-only install, with error checks
 minstall26:
 	$(call osmodlib-from-make,-C $(KERNELSRC) -p help) ; \
-	if [ -z "$$OSMODLIB" ] ; then \
+	if [ -z "$${DESTDIR}$$OSMODLIB" ] ; then \
 		$(call osmodlib-from-make,-C $(KERNELSRC) -n -p modules_install) ; \
 	fi ; \
-	if [ -z "$$OSMODLIB" ] ; then \
+	if [ -z "$${DESTDIR}$$OSMODLIB" ] ; then \
 		echo "No known place to install module. Aborting." ; \
 		exit 93 ; \
 	fi ; \
 	set -x ; \
-	mkdir -p $$OSMODLIB/kernel/$(OSMOD_DESTDIR) ; \
-	cp $(MODBUILDDIR)/ipsec.ko $$OSMODLIB/kernel/$(OSMOD_DESTDIR) ; \
+	mkdir -p $${DESTDIR}$$OSMODLIB/kernel/$(OSMOD_DESTDIR) ; \
+	cp $(MODBUILDDIR)/ipsec.ko $${DESTDIR}$$OSMODLIB/kernel/$(OSMOD_DESTDIR) ; \
 	if [ -f /sbin/depmod ] ; then \
-		/sbin/depmod -a ; \
+		/sbin/depmod -b $$DESTDIR -a $$KERNEL_VER ; \
 	fi ; \
-	if [ -n "$(OSMOD_DESTDIR)" ] ; then \
-		mkdir -p $$OSMODLIB/kernel/$(OSMOD_DESTDIR) ; \
-		if [ -f $$OSMODLIB/kernel/ipsec.ko -a -f $$OSMODLIB/kernel/$(OSMOD_DESTDIR)/ipsec.ko ] ; then \
-			echo "WARNING: two ipsec.ko modules found in $$OSMODLIB/kernel:" ; \
-			ls -l $$OSMODLIB/kernel/ipsec.ko $$OSMODLIB/kernel/$(OSMOD_DESTDIR)/ipsec.ko ; \
+	if [ -n "$${DESTDIR}$(OSMOD_DESTDIR)" ] ; then \
+		mkdir -p $${DESTDIR}$$OSMODLIB/kernel/$(OSMOD_DESTDIR) ; \
+		if [ -f $${DESTDIR}$$OSMODLIB/kernel/ipsec.ko -a -f $${DESTDIR}$$OSMODLIB/kernel/$(OSMOD_DESTDIR)/ipsec.ko ] ; then \
+			echo "WARNING: two ipsec.ko modules found in $${DESTDIR}$$OSMODLIB/kernel:" ; \
+			ls -l $${DESTDIR}$$OSMODLIB/kernel/ipsec.ko $${DESTDIR}$$OSMODLIB/kernel/$(OSMOD_DESTDIR)/ipsec.ko ; \
 			exit 1; \
 		fi ; \
 	fi
