--- opcodes/i386-dis.c	2011/08/02 19:58:06	1.265
+++ opcodes/i386-dis.c	2011/10/26 14:46:00	1.266
@@ -11331,7 +11331,7 @@
     {
       /* Too many prefixes or unused REX prefixes.  */
       for (i = 0;
-	   all_prefixes[i] && i < (int) ARRAY_SIZE (all_prefixes);
+	   i < (int) ARRAY_SIZE (all_prefixes) && all_prefixes[i];
 	   i++)
 	(*info->fprintf_func) (info->stream, "%s",
 			       prefix_name (all_prefixes[i], sizeflag));
