diff -ruN apcupsd-3.14.14.orig/doc/Makefile apcupsd-3.14.14/doc/Makefile
--- apcupsd-3.14.14.orig/doc/Makefile	2017-08-09 17:23:57.389466724 -0400
+++ apcupsd-3.14.14/doc/Makefile	2017-08-09 17:36:56.000000000 -0400
@@ -3,7 +3,7 @@
 SUBDIRS = 
 include $(topdir)/autoconf/targets.mak
 
-all-targets: apcupsd.man.txt apcaccess.man.txt apctest.man.txt \
-             apccontrol.man.txt apcupsd.conf.man.txt
+all-targets: apcupsd.man.txt apcaccess.man.txt \
+             apccontrol.man.txt
 
 apcupsd.man.txt: apcupsd.8
 	$(call MANIFY,$<,$@)
@@ -24,14 +26,14 @@
 	$(call MKDIR,$(mandir)/man8)
 	$(call INSTDATA,644,apcupsd.8,$(mandir)/man8/apcupsd.8)
 	$(call INSTDATA,644,apcaccess.8,$(mandir)/man8/apcaccess.8)
-	$(call INSTDATA,644,apctest.8,$(mandir)/man8/apctest.8)
+#	$(call INSTDATA,644,apctest.8,$(mandir)/man8/apctest.8)
 	$(call INSTDATA,644,apccontrol.8,$(mandir)/man8/apccontrol.8)
 	$(call MKDIR,$(mandir)/man5)
-	$(call INSTDATA,644,apcupsd.conf.5,$(mandir)/man5/apcupsd.conf.5)
+#	$(call INSTDATA,644,apcupsd.conf.5,$(mandir)/man5/apcupsd.conf.5)
 
 uninstall-man:
 	$(call UNINST,$(mandir)/man8/apcupsd.8)
 	$(call UNINST,$(mandir)/man8/apcaccess.8)
-	$(call UNINST,$(mandir)/man8/apctest.8)
+#	$(call UNINST,$(mandir)/man8/apctest.8)
 	$(call UNINST,$(mandir)/man8/apccontrol.8)
-	$(call UNINST,$(mandir)/man5/apcupsd.conf.5)
+#	$(call UNINST,$(mandir)/man5/apcupsd.conf.5)

